### ISSUE TITLE
Shift+Enter does not reliably switch from tracer to editor in threaded applications

### ISSUE THREAD

**Issue by mkromberg** [Dyalog/ride#626]
### Describe the issue you are having
When running a multi-threaded application, Shift+Enter in the tracer is unreliable.

### Did you connect to an already running interpreter or start the interpreter from RIDE?
Does not matter.

### How do you reproduce the issue?
```
      )load the workspace from issue 624
      Init ⋄ Server&4321
      ⎕FX 'foo' '⍝ Init
     trace foo
```
Once you are inside foo, Shift+Enter will sometimes switch from trace to edit mode - and sometimes not.

### Paste the contents of Help → About (Shift+F1)
```
IDE:
  Version: 4.4.3469
  Platform: Win32
  Date: 2020-07-07 16:02:26 +0100
  Git commit: b97d35464cceaf671e05e82e8db59e50f69a0481
  Preferences:{
    "connectFav":"1",
    "floatSingle":"0",
    "floating":"1",
    "kbdLocale":"da_DK",
    "prefixKey":"<"
  }

Interpreter:
  Version: 19.0.39694
  Platform: Linux-64
  Edition: Unicode/64
  Date: Dec  8 2020 at 03:38:37
```

**Comment by mkromberg** [comment]
Fixed with RIDE 4.4.3469 and Dyalog APL 18.1.39916

**Comment by aplteam** [comment]
Hello Gil,

That version has not been released - is this on purpose?
